docs/source/reference-core.rst

```diff
@@ -1245,58 +1245,320 @@ Broadcasting an event with :class:`Event`
+   nursery.start_soon(producer, "A", send_channel.clone())
+   nursery.start_soon(producer, "B", send_channel)
+   nursery.start_soon(consumer, "X", receive_channel.clone())
+   nursery.start_soon(consumer, "Y", receive_channel)
+
+But this is more error-prone, especially if in more complex code that
+might use a loop to spawn the producers/consumers.
+
+Just make sure that you don't write::
+
+   # Broken, will cause program to hang:
+   send_channel, receive_channel = trio.open_memory_channel(0)
+   nursery.start_soon(producer, "A", send_channel.clone())
+   nursery.start_soon(producer, "B", send_channel.clone())
+   nursery.start_soon(consumer, "X", receive_channel.clone())
+   nursery.start_soon(consumer, "Y", receive_channel.clone())
+
+Here we pass clones into the tasks, but never close the original
+objects. That means we have 3 send channel objects (the original + two
+clones), but we only close 2 of them, so the consumers will hang
+around forever waiting for that last one to be closed.
+
+
+.. _channel-buffering:
+
+Buffering in channels
++++++++++++++++++++++
+
+When you call :func:`open_memory_channel`, you have to specify how
+many values can be buffered internally in the channel. If the buffer
+is full, then any task that calls :meth:`~trio.abc.SendChannel.send`
+will stop and wait for another task to call
+:meth:`~trio.abc.ReceiveChannel.receive`. This is useful because it
+produces *backpressure*: if the channel producers are running faster
+than the consumers, then it forces the producers to slow down.
+
+You can disable buffering entirely, by doing
+``open_memory_channel(0)``. In that case any task calls
+:meth:`~trio.abc.SendChannel.send` will wait until another task calls
+`~trio.abc.SendChannel.receive`, and vice versa. This is similar to
+how channels work in the `classic Communicating Sequential Processes
+model <https://en.wikipedia.org/wiki/Channel_(programming)>`__, and is
+a reasonable default if you aren't sure what size buffer to use.
+(That's why we used it in the examples above.)
+
+At the other extreme, you can make the buffer unbounded by using
+``open_memory_channel(math.inf)``. In this case,
+:meth:`~trio.abc.SendChannel.send` *always* returns immediately.
+Normally, this is a bad idea. To see why, consider a program where the
+producer runs more quickly than the consumer::
+
+.. literalinclude:: reference-core/channels-backpressure.py
+
+If you run this program, you'll see output like:
+
+.. code-block:: none
+
+   Sent message: 0
+   Received message: 0
+   Sent message: 1
+   Sent message: 2
+   Sent message: 3
+   Sent message: 4
+   Sent message: 5
+   Sent message: 6
+   Sent message: 7
+   Sent message: 8
+   Sent message: 9
+   Received message: 1
+   Sent message: 10
+   Sent message: 11
+   Sent message: 12
+   ...
+
+On average, the producer sends ten messages per second, and the
+consumer only receives a message once per second. That means that each
```

Comment:
s/receives/reads/? To clarify that the bottleneck is not trio here, but the receiver.